### PR TITLE
chore(v1poolgrpc): changes in error handling for v1 pool operations

### PR DESCRIPTION
**Pull request by abhilashshetty04**
Added additional Error variants in lvs error enum for pool operations with more granular tonic status code. This will help Control plane to interpret error and handle accordingly.

Signed-off-by: Abhilash Shetty <abhilash.shetty@datacore.com>

### CODE REVIEW

File: io-engine/src/grpc/v0/mayastor_grpc.rs [hunk 1]
```diff
@@ -168,12 +169,27 @@ impl From<LvsError> for Status {
...
                 ..
             } => Status::invalid_argument(e.to_string()),
+
+            LvsError::PoolNotFound {
+                ..
+            } => Status::not_found(e.to_string()),
+
+            LvsError::PoolCreate {
+                source, ..
+            } => if source == Errno::EEXIST{
```
Comment (tiagolobocastro) [inline io-engine/src/grpc/v0/mayastor_grpc.rs]:
```suggestion
            } => if source == Errno::EEXIST { Status::already_exists(e.to_string()) }
                 else if if source == Errno::EINVAL { Status::invalid_argument(e.to_string()) }
                 else { Status::internal(e.to_string()) }
```

File: io-engine/src/grpc/v0/mayastor_grpc.rs [hunk 2]
```diff
@@ -168,12 +169,27 @@ impl From<LvsError> for Status {
...
             } => source.into(),
+
             LvsError::Invalid {
                 ..
             } => Status::invalid_argument(e.to_string()),
+
+            LvsError::PoolNotFound {
+                ..
+            } => Status::not_found(e.to_string()),
+
```
Comment (tiagolobocastro) [inline io-engine/src/grpc/v0/mayastor_grpc.rs]: unless you're grouping these, I'd remove the newlines between them

**Comment by tiagolobocastro** [comment]
commit message should not start with a `C`apital.

File: io-engine/src/grpc/v0/mayastor_grpc.rs (hunk 1 above)
Comment (abhilashshetty04) [reply io-engine/src/grpc/v0/mayastor_grpc.rs]: Incorporated

File: io-engine/src/grpc/v0/mayastor_grpc.rs (hunk 2 above)
Comment (abhilashshetty04) [reply io-engine/src/grpc/v0/mayastor_grpc.rs]: Incorporated

File: io-engine/src/grpc/v0/mayastor_grpc.rs [hunk 3]
```diff
@@ -164,13 +164,24 @@ impl From<LvsError> for Status {
...
             } => source.into(),
             LvsError::Invalid {
                 ..
             } => Status::invalid_argument(e.to_string()),
+            LvsError::PoolNotFound {
+                ..
+            } => Status::not_found(e.to_string()),
+            LvsError::PoolCreate {
+                source, ..
+            } => if source == Errno::EEXIST{
```
Comment (tiagolobocastro) [inline io-engine/src/grpc/v0/mayastor_grpc.rs]: nit: should have space:
```suggestion
            } => if source == Errno::EEXIST {
```
and same for EINVAL below.

**Review by tiagolobocastro — APPROVED** [review]
commit rename suggestion:
fix(v1poolgrpc): return appropriate grpc status codes

**Review by Abhinandan-Purkait — APPROVED** [review]

File: io-engine/src/grpc/v0/mayastor_grpc.rs (hunk 3 above)
Comment (abhilashshetty04) [reply io-engine/src/grpc/v0/mayastor_grpc.rs]: Incorporated

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors merge

**Pull request merged**